tfimm/architectures/convnext.py

```diff
@@ -189,7 +201,9 @@ def __init__(
         kernel_initializer, bias_initializer = _weight_initializers()
 
         self.pad = tf.keras.layers.ZeroPadding2D(padding=3)
-        self.conv_dw = tf.keras.layers.DepthwiseConv2D(
+        self.conv_dw = spectral_normalize_depthwise_conv2d(
```

Comment:
> And I would include it in the convnext.py file. And then also in resnet.py.
I did do that.

I also tried to use the function as proposed above. However, it led to ValueErrors in the tests. So, I changed back to the factory function set-up.

Reply:
Yes, you need to change
```
self.conv_dw = create_conv2d(
    depthwise=True,
    use_spec_norm=self.use_spec_norm,
    spec_norm_nb_iterations=self.spec_norm_nb_iterations,
    spec_norm_bound=self.spec_norm_bound,
    )(
    kernel_size=7,
    depthwise_initializer=kernel_initializer,
    bias_initializer=bias_initializer,
    name="conv_dw",
)
```
to
```
self.conv_dw = create_conv2d(
    depthwise=True,
    use_spec_norm=self.use_spec_norm,
    spec_norm_nb_iterations=self.spec_norm_nb_iterations,
    spec_norm_bound=self.spec_norm_bound,
    kernel_size=7,
    depthwise_initializer=kernel_initializer,
    bias_initializer=bias_initializer,
    name="conv_dw",
)
```
Then it should work.

Reply:
That will work in case of the convnext, true. but for the resnet, we are actually first creating the layer, and then call it later with different kernel sizes.

Reply:
Can we not do the same for resnet as we do for convnext, i.e., any time we need to create a conv layer, we call the `create_conv2d` function?

Reply:
Yes, I will do that.

Reply:
Ok, this one is addressed too now. resolving the thread.